src/intercom/common_redis_binding.py

```diff
@@ -1,31 +1,63 @@
+from __future__ import annotations
+
 import logging
+import os
 import pickle
-from time import time
-from typing import Any
+from multiprocessing import Process, Value
+from time import sleep, time
+from typing import TYPE_CHECKING, Any, Callable
 
 from redis.exceptions import RedisError
 
+import config
 from helperFunctions.hash import get_sha256
 from storage.redis_interface import RedisInterface
 
+if TYPE_CHECKING:
+    from objects.file import FileObject
+
 
 def generate_task_id(input_data: Any) -> str:
     serialized_data = pickle.dumps(input_data)
     return f'{get_sha256(serialized_data)}_{time()}'
 
 
-class InterComRedisInterface:
-    def __init__(self):
-        self.redis = RedisInterface()
+def publish_available_analysis_plugins(plugin_dict: dict[str, tuple]):
+    redis = RedisInterface()
+    redis.set('analysis_plugins', plugin_dict)
 
 
-class InterComListener(InterComRedisInterface):
+class InterComListener:
     """
     InterCom Listener Base Class
     """
 
     CONNECTION_TYPE = 'test'  # unique for each listener
 
+    def __init__(self, processing_function: Callable[[FileObject], None] | None = None):
+        super().__init__()
+        self.redis = RedisInterface()
+        self.process = None
+        self.processing_function = processing_function
+        self.stop_condition = Value('i', 0)
+
+    def start(self):
+        self.process = Process(target=self._worker)
+        self.process.start()
+
+    def shutdown(self):
+        self.stop_condition.value = 1
```

Comment:
I guess this plays into the _make data flow comprehensible_ goal, but this does create another dozen or so file descriptors compared to sharing one stop condition over all listeners, right?

Reply:
I also wondered about that and thought about using one central value, that is passed to the individual listeners, but the `multiprocessing.Value` objects by themselves do not seem to use additional file descriptors